src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions/src/ConsolidatedMvcViewDocumentClassifierPass.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Intermediate;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Extensions
+{
+    public class ConsolidatedMvcViewDocumentClassifierPass : DocumentClassifierPassBase
+    {
+        public static readonly string MvcViewDocumentKind = "mvc.1.0.view";
+
+        protected override string DocumentKind => MvcViewDocumentKind;
+
+        private RazorLanguageVersion _razorLanguageVersion;
+
+        public ConsolidatedMvcViewDocumentClassifierPass(RazorLanguageVersion razorLanguageVersion)
+        {
+            _razorLanguageVersion = razorLanguageVersion;
+        }
+
+        // ConsolidatedMvcViewDocumentClassifier should not apply for design time
+        // builds or language versions less than the most recent. This feature is
+        // intended to be used alongside source generator support in the Razor compiler.
+        protected override bool IsMatch(RazorCodeDocument codeDocument, DocumentIntermediateNode documentNode)
+        {
+            return documentNode.Options.DesignTime != true && _razorLanguageVersion == RazorLanguageVersion.Latest;
```

Comment:
Right now RazorLangVersion.Latest = 5.0 which would mean this applies to .net5.0 apps. I think you would need to invent a new version. Alternatively, could we add an option in `RazorExtensions.Register` that causes this type to be used?